### PR TITLE
nrf_modem: release v2.6.0

### PR DESCRIPTION
**Pull request by lemrey**
For a detailed list of changes, see the CHANGELOG file.

sdk-nrf PR: https://github.com/nrfconnect/sdk-nrf/pull/14173

### CODE REVIEW

File: nrf_modem/doc/at_interface.rst [hunk 1]
```diff
@@ -219,6 +219,38 @@ In this case, the application need not provide any intermediate buffers and can
...
+************
+
+The Modem library allows the application to be notified on functional mode changes in the modem by adding a CFUN handler through the :c:func:`nrf_modem_at_cfun_handler_set` function.
+The handler is called after an CFUN AT command is successfully processed by the modem.
+
+.. note::
+   The CFUN handler is not supported with :c:func:`nrf_modem_at_cmd_async`.
+
+.. note::
+   After the CFUN handler is done processing the event, it must call :c:func:`nrf_modem_at_cfun_handler_completed` to notify the modem library that the event is complete.
```
Comment (tokangas) [inline nrf_modem/doc/at_interface.rst:232]: Is this note correct? I don't see this function anywhere.

**Comment by jfischer-no** [comment]
@lemrey Please get some +1 asap.

**Comment by jfischer-no** [comment]
@nrfconnect/ncs-doc-owners Please take a look.

**Comment by divipillai** [comment]
> @nrfconnect/ncs-doc-owners Please take a look.

I am reviewing the PR.

**Review by SeppoTakalo — APPROVED** [review]

**Comment by jfischer-no** [comment]
@nrfconnect/ncs-doc-owners okay to merge it?

**Comment by eivindj-nordic** [comment]
> @nrfconnect/ncs-doc-owners okay to merge it?

No, tests are failing in https://github.com/nrfconnect/sdk-nrf/pull/14173

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 2]
```diff
@@ -125,7 +125,7 @@ NRF_SO_REUSEADDR
 NRF_SO_RCVTIMEO
    Set a timeout value for the :c:func:`nrf_recv`, :c:func:`nrf_recvfrom` and :c:func:`nrf_accept` operations.
    This option accepts an :c:struct:`nrf_timeval` structure with a number of seconds and microseconds specifying the limit on how long to wait for an input operation to complete.
-   If a receive operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to :c:macro:`NRF_EAGAIN` or :c:macro:`NRF_EWOULDBLOCK` if no data were received.
+   If a receive operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to ``NRF_EAGAIN`` or ``NRF_EWOULDBLOCK``if no data were received.
```
Comment (MirkoCovizzi) [inline nrf_modem/doc/sockets/socket_options_func.rst:128]:
```suggestion
   If a receive operation has blocked for this much time without receiving additional data, it returns with a partial count, or ``errno`` is set to ``NRF_EAGAIN`` or ``NRF_EWOULDBLOCK`` if no data were received.
```

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 3]
```diff
@@ -199,11 +199,11 @@ NRF_SO_RAI
...
-    * :c:macro:`NRF_RAI_LAST` - Enter RRC idle mode after the next output operation on this socket is complete.
-    * :c:macro:`NRF_RAI_ONE_RESP` - After the next output operation is complete, wait for one more packet to be received from the network on this socket before entering RRC idle mode.
-    * :c:macro:`NRF_RAI_ONGOING` - Keep RRC in connected mode after the next output operation on this socket (client side).
-    * :c:macro:`NRF_RAI_WAIT_MORE` - Keep RRC in connected mode after the next output operation on this socket (server side).
+    * ``NRF_RAI_NO_DATA`` - Immediately enter RRC idle mode for this socket. Does not require a following output operation.
+    * ``NRF_RAI_LAST`` - Enter RRC idle mode after the next output operation on this socket is complete.
+    * ``NRF_RAI_ONE_RESP`` - After the next output operation is complete, wait for one more packet to be received from the network on this socket before entering RRC idle mode.
+    * ``NRF_RAI_ONGOING`` - Keep RRC in connected mode after the next output operation on this socket (client side).
+    * ``NRF_RAI_WAIT_MORE``- Keep RRC in connected mode after the next output operation on this socket (server side).
```
Comment (MirkoCovizzi) [inline nrf_modem/doc/sockets/socket_options_func.rst:206]:
```suggestion
    * ``NRF_RAI_WAIT_MORE`` - Keep RRC in connected mode after the next output operation on this socket (server side).
```

File: nrf_modem/doc/CHANGELOG.rst [hunk 4]
```diff
@@ -15,53 +15,44 @@ nrf_modem 2.6.0
...
-  * Fixed a bug where the :c:macro:`NRF_POLLNVAL` event would not be reported when using poll callbacks set with the :c:macro:`NRF_SO_POLLCB` socket option.
-  * Fixed a bug where retrieving the value of :c:macro:`NRF_SO_SEC_HOSTNAME` caused a bad memory access, if the option had not been set.
   * The :c:macro:`nrf_sa_family_t` type definition to ``unsigned short`` (from ``unsigned int``), to reduce the size of the socket address types.
   * The type of the field :c:member:`nrf_sockaddr.sa_family` to :c:macro:`nrf_sa_family_t` (from ``int``).
   * The type of the field :c:member:`nrf_sockaddr_in6.sin6_scope_id` to ``uint8_t`` (from ``uint32_t``).
+  * The :c:macro:`NRF_SO_RCVTIMEO` socket option can now be used to set a timeout for the :c:func:`nrf_accept` operation.
+
+* Fixed:
+
+  * Rare multi-threading bugs in the :c:func:`nrf_socket`, :c:func:`nrf_recv` and :c:func:`nrf_connect` functions.
```
Comment (MirkoCovizzi) [inline nrf_modem/doc/CHANGELOG.rst:39]: Missing comma.
```suggestion
  * Rare multi-threading bugs in the :c:func:`nrf_socket`, :c:func:`nrf_recv`, and :c:func:`nrf_connect` functions.
```

Comment (eivindj-nordic) [reply nrf_modem/doc/CHANGELOG.rst:39]: I don't think this should be here.

**Review by MirkoCovizzi — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Comment by jfischer-no** [comment]
@eivindj-nordic ready?

**Comment by eivindj-nordic** [comment]
> @eivindj-nordic ready?

Waiting for carrier lib to be ready in https://github.com/nrfconnect/sdk-nrf/pull/14195.

**Pull request merged**